nf_core/components/components_utils.py

```diff
@@ -159,10 +156,38 @@ def get_components_to_install(subworkflow_dir: Union[str, Path]) -> Tuple[List[s
                 name, link = match.groups()
                 if link.startswith("../../../"):
                     name_split = name.lower().split("_")
-                    modules.append("/".join(name_split))
+                    component_name = "/".join(name_split)
+                    component_dict: Dict[str, Optional[str]] = {
+                        "name": component_name,
+                    }
+                    modules[component_name] = component_dict
                 elif link.startswith("../"):
-                    subworkflows.append(name.lower())
-    return modules, subworkflows
+                    component_name = name.lower()
+                    component_dict = {"name": component_name}
+                    subworkflows[component_name] = component_dict
+
+    if (sw_meta := Path(subworkflow_dir, "meta.yml")).exists():
+        with open(sw_meta) as fh:
+            meta = yaml.safe_load(fh)
+            if "components" in meta:
+                components = meta["components"]
+                for component in components:
+                    if isinstance(component, dict):
+                        component_name = list(component.keys())[0].lower()
+                        branch = component[component_name].get("branch")
+                        git_remote = component[component_name]["git_remote"]
+                        modules_repo = ModulesRepo(git_remote, branch=branch)
+                        current_comp_dict = subworkflows if component_name in subworkflows else modules
+
+                        component_dict = {
+                            "org_path": modules_repo.repo_path,
+                            "git_remote": git_remote,
+                            "branch": branch,
+                        }
+
+                        current_comp_dict[component_name].update(component_dict)
```

Comment:
if I understand this correctly, if have an nf-core module and want to install the same module with a module from a different org, this would just silently overwrite it. it should at least log a warning, but better ask first before overwriting.

Reply:
You should (in-theory) be able to install two same-named modules under different orgs. I haven't checked if that is true for this new code, but it's a pattern that I always intended and expected the private remotes code to be able to handle

Reply:
okay, makes sense.

Reply:
I *think* after looking at this more that the issue will be that you cannot use two modules with the same name from two repos, in the same subworkflow. 

i.e. this would break on install:

```
include {MINIMAP2_ALIGN} from '../modules/nf-core/minimap2/align'
include {MINIMAP2_ALIGN as CUSTOM_MINIMAP2} from '../modules/my_org/minimap2/align'
```

Reply:
I thought this was covered by one of the tests, but I'm not 100% sure. @jvfe can comment.

Reply:
indeed, it doesn't override the module, but you are able to install the same module from two different repos 👍